KStudy.js

```diff
@@ -0,0 +1,509 @@
+
+	KISS (Korean Studies Information Service System) Translator
+	Copyright © 2017-2018 Yunwoo Song, Frank Bennett, and Philipp Zumstein
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (url.includes('/thesis/thesis-view.asp')) {
+		return "journalArticle";
+	} else if (url.includes('/public2-article.asp')) {
+		// these are reports and working paper series but with publicaton name,
+		// volume, issue numbers; thus handled as journal articles as well
+		return "journalArticle";
+	} else if (url.includes('/public3-article.asp')) {
+		return "report";
+	} else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//div[contains(@class, "thesis-info")]/h5/a');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		// exclude no real links
+		if (href=="#") continue;
+		// exclude links to overview of journal
+		if (href.includes("/journal/journal-view")) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var key = url.match(/\bkey=(\d+)\b/)[1];
+	var risURL = "http://kiss.kstudy.com/p-common/export_endnote.asp";
+	var postData = "atcl_data=" + key + "&export_gubun=EndNote";
+	ZU.doPost(risURL, postData, function(text) {
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+		translator.setString(text);
+		translator.setHandler("itemDone", function(obj, item) {
+			if (url.includes('/public/public3-article')) {
+				item.itemType = "report";
+			}
+			if (item.issue && item.issue == "0") {
+				delete item.issue;
+			}
+			item.language = "ko-KR";
+			if (item.pages) {
+				item.pages = item.pages.replace('-', '–');
```

Comment:
we don't usually do this -- do we actually prefer en-dashes over hyphens in the database?

Reply:
I don't think there's any harm in doing so, though we should probably just be automatically converting hyphens in page ranges to en dashes on save.

Reply:
Hm... my mistake then. How is the page range then shown in citation styles? Maybe the replacement could also happens at some later step... I will delete it on the translator code, beause this would then be a larger issue.